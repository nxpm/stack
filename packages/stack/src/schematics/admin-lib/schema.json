{
  "$schema": "http://json-schema.org/draft-07/schema",
  "id": "AdminLib",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the library?",
      "pattern": "^[a-zA-Z]{1}.*$"
    },
    "type": {
      "description": "The type of library",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 1
      },
      "x-prompt": {
        "message": "Which type of library would you like to create?",
        "type": "list",
        "items": [
          {
            "value": "assets",
            "label": "Assets library"
          },
          {
            "value": "data-access",
            "label": "Data access library"
          },
          {
            "value": "feature",
            "label": "Feature library"
          },
          {
            "value": "style",
            "label": "Style Library"
          },
          {
            "value": "ui",
            "label": "UI Library"
          },
          {
            "value": "util",
            "label": "Utility library"
          }
        ]
      }
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the project (used for linting)",
      "alias": "t"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed",
      "alias": "d"
    },
    "publishable": {
      "type": "boolean",
      "default": false,
      "description": "Generate a publishable library."
    },
    "buildable": {
      "type": "boolean",
      "default": false,
      "description": "Generate a buildable library."
    },
    "prefix": {
      "type": "string",
      "format": "html-selector",
      "description": "The prefix to apply to generated selectors.",
      "alias": "p"
    }
  },
  "required": ["name"]
}
