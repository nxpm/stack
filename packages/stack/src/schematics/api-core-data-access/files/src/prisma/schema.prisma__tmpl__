datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider      = "prisma-client-js"
  binaryTargets = ["native"]
}

model User {
  id                   String           @id @default(cuid())
  createdAt            DateTime         @default(now())
  updatedAt            DateTime         @default(now()) @updatedAt
  role                 Role
  developer            Boolean          @default(false)
  email                String           @unique
  username             String           @unique
  password             String?
  firstName            String?
  lastName             String?
  avatarUrl            String?
  location             String?
  phone                String?
  bio                  String?
}

enum Role {
  Admin
  User
}

