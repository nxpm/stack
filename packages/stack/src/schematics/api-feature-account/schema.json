{
  "$schema": "http://json-schema.org/draft-07/schema",
  "id": "ApiFeatureUser",
  "title": "",
  "type": "object",
  "properties": {
    "appName": {
      "type": "string",
      "description": "Name of the app to connect to",
      "alias": "a"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed",
      "alias": "d"
    }
  },
  "required": []
}
