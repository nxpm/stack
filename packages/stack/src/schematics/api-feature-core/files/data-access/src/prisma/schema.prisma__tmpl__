datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider      = "prisma-client-js"
  binaryTargets = ["native"]
}

model User {
  id        String   @id @default(cuid())
  createdAt DateTime @default(now())
  updatedAt DateTime @default(now()) @updatedAt
  role      Role
  developer Boolean  @default(false)
  username  String   @unique
  password  String?
  firstName String?
  lastName  String?
  avatarUrl String?
  location  String?
  phone     String?
  bio       String?
  emails    Email[]  @relation("Email_belongsTo_Owner")
}

enum Role {
  Admin
  User
}

model Email {
  id            String    @id @default(cuid())
  createdAt     DateTime  @default(now())
  updatedAt     DateTime  @default(now()) @updatedAt
  email         String    @unique
  public        Boolean   @default(false)
  primary       Boolean   @default(false)
  verified      Boolean   @default(false)
  verifyToken   String?
  verifyExpires DateTime?
  owner         User      @relation(name: "Email_belongsTo_Owner", fields: [ownerId], references: [id])
  ownerId       String
  // Enable when this is resolved https://github.com/prisma/prisma/issues/3387
  // @@unique([ownerId, primary])
}
