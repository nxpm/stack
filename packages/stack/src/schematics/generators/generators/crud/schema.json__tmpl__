{
  "cli": "nx",
  "id": "web-module",
  "type": "object",
  "properties": {
    "target": {
      "type": "string",
      "description": "Target folder to create the module",
      "$default": {
        "$source": "argv",
        "index": 0
      }
    },
    "name": {
      "type": "string",
      "description": "Name of the module (in kebab-case)",
      "$default": {
        "$source": "argv",
        "index": 1
      }
    }
  },
  "required": ["target", "name"]
}
