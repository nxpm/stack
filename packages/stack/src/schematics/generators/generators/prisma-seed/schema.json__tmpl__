{
  "cli": "nx",
  "id": "prisma-seed",
  "type": "object",
  "properties": {
    "confirm": {
      "type": "boolean",
      "description": "Library confirm",
      "x-prompt": "Are you sure you want to delete and seed the database?"
    },
    "timeout": {
      "type": "number",
      "description": "Number of seconds to wait before before starting deletion",
      "default": 10
    }
  },
  "required": ["confirm", "timeout"]
}
