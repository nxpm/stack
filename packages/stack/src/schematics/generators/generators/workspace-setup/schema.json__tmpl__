{
  "cli": "nx",
  "id": "workspace-setup",
  "type": "object",
  "properties": {},
  "required": []
}
