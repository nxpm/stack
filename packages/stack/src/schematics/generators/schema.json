{
  "$schema": "http://json-schema.org/draft-07/schema",
  "id": "Init",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "default": "web"
    }
  },
  "required": []
}
