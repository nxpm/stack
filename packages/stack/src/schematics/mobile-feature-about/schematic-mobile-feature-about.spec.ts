import { Tree } from '@angular-devkit/schematics'
import { SchematicTestRunner } from '@angular-devkit/schematics/testing'
import { createEmptyWorkspace } from '@nrwl/workspace/testing'
import { join } from 'path'

import { MobileFeatureAboutSchematicSchema } from './schema'

describe('mobile-feature-about schematic', () => {
  let appTree: Tree
  const options: MobileFeatureAboutSchematicSchema = { appName: 'test' }

  const testRunner = new SchematicTestRunner('@nxpm/mobile-feature-about', join(__dirname, '../../../collection.json'))

  beforeEach(() => {
    appTree = createEmptyWorkspace(Tree.empty())
  })

  it('should run successfully', async () => {
    await expect(
      testRunner.runSchematicAsync('mobile-feature-about', options, appTree).toPromise(),
    ).resolves.not.toThrowError()

    expect(appTree.read('libs/test/about/feature/src/lib/test-about-feature.module.ts').toString()).toMatchSnapshot()
  })
})
