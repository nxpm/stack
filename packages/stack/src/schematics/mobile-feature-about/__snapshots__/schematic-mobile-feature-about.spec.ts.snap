// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mobile-feature-about schematic should run successfully 1`] = `
"import { NgModule } from '@angular/core'
import { CommonModule } from '@angular/common'
import { RouterModule } from '@angular/router'
import { TestUiPageModule } from '@nxpm/test/ui/page'
import { TestAboutFeatureComponent } from './test-about-feature.component'

@NgModule({
  declarations: [TestAboutFeatureComponent],
  imports: [
    CommonModule,
    RouterModule.forChild([{ path: '', pathMatch: 'full', component: TestAboutFeatureComponent }]),
    TestUiPageModule,
  ],
})
export class TestAboutFeatureModule {}
"
`;
